### ISSUE TITLE
Nest app failing to build when attempting to generate package.json

### ISSUE THREAD

**Issue by joshdawson** [nrwl/nx#22215]
### Current Behavior

Build is failing for Nest app after adding in option to generate `package.json`. I'm wanting to generate the `package.json` so I can dockerize my app which currently sits in an Nx monorepo.

What's interesting is that despite the build failing, there are built files in the `dist` folder that run absolutely fine. As in if I run `node dist/{my-app}/main.js` it works as expected.

### Expected Behavior

Build should pass and `package.json` and `package-lock.json` should be generated

### Nx Report

```shell
Node   : 20.11.0
OS     : linux-x64
npm    : 10.2.4

nx (global)        : 18.0.7
nx                 : 18.0.7
@nx/js             : 18.0.7
@nx/jest           : 18.0.7
@nx/linter         : 18.0.7
@nx/eslint         : 18.0.7
@nx/workspace      : 18.0.7
@nx/devkit         : 18.0.7
@nx/eslint-plugin  : 18.0.7
@nx/nest           : 18.0.7
@nx/node           : 18.0.7
@nrwl/tao          : 18.0.7
@nx/web            : 18.0.7
@nx/webpack        : 18.0.7
typescript         : 5.3.3
```

### Failure Logs

```shell
> nx run toon-library-api:build

assets by path *.json 12.9 KiB
  asset package-lock.json 12.4 KiB [emitted]
  asset package.json 509 bytes [emitted]
asset main.js 14.3 KiB [emitted] (name: main)
asset assets/.gitkeep 0 bytes [emitted] [from: apps/toon-library-api/src/assets/.gitkeep] [copied]
cacheable modules 11.4 KiB
  modules by path ./libs/ 5.16 KiB 7 modules
  modules by path ./apps/toon-library-api/src/ 6.29 KiB
    modules by path ./apps/toon-library-api/src/api/*.ts 4.8 KiB 4 modules
    + 2 modules
modules by path external "@nestjs/ 210 bytes
  external "@nestjs/common" 42 bytes [built] [code generated]
  external "@nestjs/core" 42 bytes [built] [code generated]
  external "@nestjs/apollo" 42 bytes [built] [code generated]
  + 2 modules
external "tslib" 42 bytes [built] [code generated]
external "path" 42 bytes [built] [code generated]
external "typeorm" 42 bytes [built] [code generated]
external "dotenv/config" 42 bytes [built] [code generated]
external "envalid" 42 bytes [built] [code generated]
webpack 5.90.3 compiled successfully in 1767 ms

 NX   The "paths[1]" argument must be of type string. Received undefined


TypeError [ERR_INVALID_ARG_TYPE]: The "paths[1]" argument must be of type string. Received undefined
    at validateString (node:internal/validators:162:11)
    at resolve (node:path:1101:7)
    at /home/josh/dev/personal/toon-library-api/node_modules/@nx/webpack/src/executors/webpack/webpack.impl.js:119:41
    at /home/josh/dev/personal/toon-library-api/node_modules/rxjs/dist/cjs/internal/operators/switchMap.js:17:35
    at OperatorSubscriber._this._next (/home/josh/dev/personal/toon-library-api/node_modules/rxjs/dist/cjs/internal/operators/OperatorSubscriber.js:33:21)
    at Subscriber.next (/home/josh/dev/personal/toon-library-api/node_modules/rxjs/dist/cjs/internal/Subscriber.js:51:18)
    at /home/josh/dev/personal/toon-library-api/node_modules/rxjs/dist/cjs/internal/operators/bufferCount.js:52:36
    at OperatorSubscriber._this._next (/home/josh/dev/personal/toon-library-api/node_modules/rxjs/dist/cjs/internal/operators/OperatorSubscriber.js:33:21)
    at Subscriber.next (/home/josh/dev/personal/toon-library-api/node_modules/rxjs/dist/cjs/internal/Subscriber.js:51:18)
    at /home/josh/dev/personal/toon-library-api/node_modules/rxjs/dist/cjs/internal/operators/mergeInternals.js:28:28

———————————————————————————————————————————————————————————————————————————————————————————————————————————————————————

 NX   Running target build for project toon-library-api failed

Failed tasks:

- toon-library-api:build
```

### Operating System

- [ ] macOS
- [X] Linux
- [ ] Windows
- [ ] Other (Please specify)

### Additional Information

Build worked fine before trying to generate the `package.json`

**Comment by joshdawson** [comment]
Closing this as I've found an alternative way to solve the problem I was haviing

**Comment by dan-cooke** [comment]
@joshdawson  care to elaborate? I am experiencing this exact issue

**Comment by joshdawson** [comment]
> @joshdawson care to elaborate? I am experiencing this exact issue

Sure. As I mentioned in the issue description, I was wanting to generate the `package.json` for the app for CI.

I managed to get the build to work by adding effectively a dummy `package.json` and `package-lock.json` to the root folder of the app that I wanted the `package.json` to be generated.

Essentially
```json
{
  "name": "my-app",
  "version": "0.0.1",
  "dependencies": {
    "tslib": "^2.3.0"
  },
  "type": "commonjs",
  "main": "./src/index.js",
  "typings": "./src/index.d.ts"
}
```
And then a `package-lock.json` that mirrored that. The build process then replaces these dummy files with the correct versions.

The only change I made to the build configuration was to add `generatePackageJson: true` in the `webpack.config.js` of the app.
